prov/efa/src/efa_cq.h

```diff
@@ -3,6 +3,77 @@
 
 #include "efa.h"
 
+enum ibv_cq_ex_type {
+	IBV_CQ,
+	EFADV_CQ
+};
+
+struct efa_ibv_cq {
+	struct ibv_cq_ex *ibv_cq_ex;
+	enum ibv_cq_ex_type ibv_cq_ex_type;
+};
+
+struct efa_ibv_cq_poll_list_entry {
+	struct dlist_entry	entry;
+	struct efa_ibv_cq	*cq;
+};
+
+void efa_rdm_cq_poll_ibv_cq(ssize_t cqe_to_process, struct efa_ibv_cq *ibv_cq);
+
+static inline
+int efa_ibv_cq_poll_list_match(struct dlist_entry *entry, const void *cq)
+{
+	struct efa_ibv_cq_poll_list_entry *item;
+	item = container_of(entry, struct efa_ibv_cq_poll_list_entry, entry);
+	return (item->cq == cq);
+}
+
+
+static inline
+int efa_ibv_cq_poll_list_insert(struct dlist_entry *poll_list, struct ofi_genlock *lock, struct efa_ibv_cq *cq)
+{
+	int ret = 0;
+	struct dlist_entry *entry;
+	struct efa_ibv_cq_poll_list_entry *item;
+
+	ofi_genlock_lock(lock);
+	entry = dlist_find_first_match(poll_list, efa_ibv_cq_poll_list_match, cq);
+	if (entry) {
+		ret = -FI_EALREADY;
+		goto out;
+	}
+
+	item = calloc(1, sizeof(*item));
```

Comment:
Is `calloc()` necessary over regular `malloc()`?

Reply:
I don't think it's necessary, I followed other similar functions that allocate the list entry.